charts/argo-cd/templates/argocd-application-controller/deployment.yaml

```diff
@@ -14,7 +14,7 @@ metadata:
 spec:
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "argo-cd.name" . }}-application-controller
+      app.kubernetes.io/name: {{ include "argo-cd.name" . }}-{{ .Values.controller.name }}
```

Comment:
It looks like this change will break upgrades of the chart as the `selector` field is [immutable](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates).

Reply:
Given the current value is `application-controller` then this shouldn't break existing upgrades but a README.md callout would be good.

Reply:
Yep! already the default value of `.Values.controller.name` is `application-controller`